src/frequenz/dispatch/_dispatcher.py

```diff
@@ -107,34 +178,69 @@ def __init__(
             grpc_channel: The gRPC channel.
             svc_addr: The service address.
         """
-        self._ready_channel = Broadcast[Dispatch](name="ready_dispatches")
-        self._updated_channel = Broadcast[DispatchEvent](name="new_dispatches")
+        self._running_state_channel = Broadcast[Dispatch](name="running_state_change")
+        self._lifecycle_events_channel = Broadcast[DispatchEvent](
+            name="lifecycle_events"
+        )
+        self._client = Client(grpc_channel, svc_addr)
         self._actor = DispatchingActor(
             microgrid_id,
-            grpc_channel,
-            svc_addr,
-            self._updated_channel.new_sender(),
-            self._ready_channel.new_sender(),
+            self._client,
+            self._lifecycle_events_channel.new_sender(),
+            self._running_state_channel.new_sender(),
         )
 
     async def start(self) -> None:
         """Start the actor."""
         self._actor.start()
 
+    @property
+    def client(self) -> Client:
+        """Return the client."""
+        return self._client
+
     @property
     def lifecycle_events(self) -> ReceiverFetcher[DispatchEvent]:
-        """Return new, updated or deleted dispatches receiver.
+        """Return new, updated or deleted dispatches receiver fetcher.
 
         Returns:
             A new receiver for new dispatches.
         """
-        return self._updated_channel
+        return self._lifecycle_events_channel
 
     @property
-    def ready_to_execute(self) -> ReceiverFetcher[Dispatch]:
-        """Return ready dispatches receiver.
+    def running_status_change(self) -> ReceiverFetcher[Dispatch]:
```

Comment:
Active seems wrong as well, given that it also sends deleted and specifically inactive dispatches, too...

Reply:
> like which types of dispatch objects are put here. I think currently running dispatches are put here, or active dispatches.

Any "type" that makes a receiver start, stop or reconfigure.

> So what about either just active, or currently_active?

It includes deleted, inactive and finished dispatches, as well. 

> or we call it something like _events like lifecycle_events 

We could call it `running_status_events` :shrug: 

> and wrap the dispatches inside some other object that tells which kind of update is it.

Not sure what help that is, there is no specific "kind" or "type" of update.
Well, or if you want, you could categorize it as

1) request to start
2) request to stop
3) request to use different configuration

all those cases are covered by the simple usage example already shown, basically for 1) and 3) you check if you're running, if yes, reconfigure, if no, start. (or stop if it's 2)

Reply:
> It includes deleted, inactive and finished dispatches, as well.

Yeah, but considering both the previous and new state, then the dispatch was active (either it needs to be active now, or it was active when the change was received). It is a broad definition of active but I think it could still work given that it is very complex what we are trying to describe and no sensible symbol name will be able to fully describe it.

> We could call it `running_status_events` 🤷

Yeah, sure, but in that case I don't think we should stream `Dispatch` objects, we should stream some `RunningStatusEvent`, which makes the interface more complicated, because what do you put there?

> Not sure what help that is, there is no specific "kind" or "type" of update. Well, or if you want, you could categorize it as
> 
>     1. request to start
> 
>     2. request to stop
> 
>     3. request to use different configuration

Mmmm, maybe. For some reason I'm not convinced to add so much meaning to an update. What we are really receiving is an update to the dispatch, which could end up in those 3 points or not. But I don't know exactly why I don't like this mapping, maybe is just a general feeling of leaking abstractions, maybe you are right and this approach is OK.

But then, when the dispatch finishes normally you won't get a notification this way, right? Or will you? I thought we only send something here if the duration changed, or it was disabled, or something else that makes the dispatch not be active anymore, but users should still make sure the dispatch is stopped after the duration even if they don't receive any notification, right? My fear here is that if users believe they will always be notified about a stopping event, then they will not look into the duration anymore.

Actually thinking about it, the actor could send a stop request and keep the timer(s) itself to trigger the event after the duration. I guess the only risk here is the dispatch actor crashing and the users not being notified about the stop. We could potentially handle this correctly and recover the timers after the crash, so only an infinite restart loop could a real problem, but maybe in that case is reasonable to assume everything went to hell.

Another issue with handling the duration timer in the dispatch actor is that we'll have to set a timer for every dispatch, even if nobody cares about it. Let's say for example a FCR dispatch has an invalid payload parameter, so we don't run it, we'll still have a timer in the dispatch actor for it. If we combine it with the feature of automatically filtering by type in the `Dispatcher`, then I guess this problem is not that bad, as it should be rare that someone requested for a type of Dispatch and then they don't really run it.

OK, if we go this route I think it could work: We have `running_status_events`, which streams something like `Start | Stop | Update` events, we provide type filtering via the `new_receiver()` and we make the actor handle the dispatch timer. This will make the dispatch actor much more complex, but I guess it might be worth it if we remove that complexity from the user...